scout/server/blueprints/alignviewers/controllers.py

```diff
@@ -1,15 +1,69 @@
 # -*- coding: utf-8 -*-
 import logging
+import os.path
 from flask_login import current_user
 from scout.server.extensions import store, cloud_tracks
 from scout.constants import IGV_TRACKS, CASE_SPECIFIC_TRACKS
+from scout.server.utils import institute_and_case
 
 LOG = logging.getLogger(__name__)
 CUSTOM_TRACK_NAMES = ["Genes", "ClinVar", "ClinVar CNVs"]
 
 
+def make_sashimi_tracks(institute_id, case_name, variant_id, build="38"):
+    """Create a dictionary containing the required tracks for a splice junction plot
+
+    Accepts:
+        institute_id(str): institute _id
+        case_name(str): case display name
+        variant_id(str) _id of a variant
+    Returns:
+        display_obj(dict): A display object containing case name, list of genes, lucus and tracks
+    """
+    # Collect locus coordinates. Take into account that variant can hit multiple genes
+    variant_obj = store.variant(document_id=variant_id)
+    variant_genes_ids = [gene["hgnc_id"] for gene in variant_obj.get("genes", [])]
+    locus_start_coords = []
+    locus_end_coords = []
+    gene_symbols = []
+    for gene_id in variant_genes_ids:
+        gene_obj = store.hgnc_gene(hgnc_identifier=gene_id, build=build)
+        if gene_obj is None:
+            continue
+        gene_symbols.append(gene_obj.get("hgnc_symbol") or gene_obj["hgnc_id"])
+        locus_start_coords.append(gene_obj["start"])
+        locus_end_coords.append(gene_obj["end"])
+
+    locus_start = min(locus_start_coords)
+    locus_end = max(locus_end_coords)
+    locus = f"{variant_obj['chromosome']}:{locus_start}-{locus_end}"  # Locus will span all genes the variant falls into
+    display_obj = {"locus": locus, "tracks": [], "genes": gene_symbols}
+
+    # Populate tracks for each individual with splice junction track data
+    _, case_obj = institute_and_case(store, institute_id, case_name)
+    for ind in case_obj.get("individuals", []):
+        if not all([ind.get("splice_junctions_bed"), ind.get("rna_coverage_bigwig")]):
+            continue
+
+        coverage_wig = ind["rna_coverage_bigwig"]
+        splicej_bed = ind["splice_junctions_bed"]
+        splicej_bed_index = f"{splicej_bed}.tbi" if os.path.isfile(f"{splicej_bed}.tbi") else None
```

Comment:
Display will fail if the index is missing, but silently. I guess the target page has no flash, and it's perhaps not end user actionable info anyway: perhaps log something so it's apparent things are not quite well with the case?

Reply:
Also note the index naming convention in docs somewhere!

Reply:
It will work without an index, I tried. Perhaps it will be a bit slower. But an alert would be nice to have, I agree!

Reply:
Interesting; mine didn't - perhaps some library version differences or browser timeout thresholds somewhere? Anyways, even better, but a small alert here wouldn't hurt - we are verbose enough with other stuff.. 😝

Reply:
I've tried not with the files downloadable from figshare, but some complete profiles from a real analysis. Could be that affects the result as well!

Reply:
fixed:

![image](https://user-images.githubusercontent.com/28093618/114022447-11f69b00-9872-11eb-9c80-53bd4e938b67.png)